src/api/Feed.js

```diff
@@ -228,6 +234,24 @@ class Feed extends Base {
         });
     }
 
+    /**
+     * Fetches the current version for a file
+     *
+     * @return {Promise} - the file versions
+     */
+    fetchCurrentVersion(file: BoxItem): Promise<?FileVersions> {
+        this.versionsAPI = new VersionsAPI(this.options);
+
+        return new Promise(resolve => {
+            this.versionsAPI.getCurrentVersion(
+                this.id,
+                this.versionId,
+                currentVersion => resolve(this.versionsAPI.decorateCurrentVersion(currentVersion, file)),
```

Comment:
You may be able to move this method call to a `.then` after the promise instantiation.